sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -95,18 +95,21 @@ abstract class QueryStageExec extends LeafExecNode {
   /**
    * Compute the statistics of the query stage if executed, otherwise None.
    */
-  def computeStats(): Option[Statistics] = resultOption.get().map { _ =>
+  def computeStats(): Option[Statistics] = if (isMaterialized) {
     val runtimeStats = getRuntimeStatistics
     val dataSize = runtimeStats.sizeInBytes.max(0)
     val numOutputRows = runtimeStats.rowCount.map(_.max(0))
-    Statistics(dataSize, numOutputRows, isRuntime = true)
+    Some(Statistics(dataSize, numOutputRows, isRuntime = true))
+  } else {
+    None
   }
 
   @transient
   @volatile
   protected var _resultOption = new AtomicReference[Option[Any]](None)
 
   private[adaptive] def resultOption: AtomicReference[Option[Any]] = _resultOption
+  private[adaptive] def isMaterialized: Boolean = resultOption.get().isDefined
```

Comment:
`materialize` is a public method, I think it makes more sense to make `isMaterialized` public as well.

Reply:
I'm ok with it,  but just wonder which code place may use `isMaterialized` without adaptive ?